### DISCUSSION TITLE
Command (or way) to toggle a views "container" on and off?

### DISCUSSION THREAD

**Discussion by eamodio** [microsoft/vscode-discussions#547] [Extension Development QnA] [answered]
Anyone know of a way to toggle on/off the "container" (sidebar, panel, secondary sidebar) where a custom view lives? Some of the built-in views have this ability (terminal, debug, output), but I haven't found a way to do this from an extension.

I can toggle each container on/off directly, but there isn't a way (AKAIK) to tell what container your view is in, nor a way to toggle your view's "container"

For example, when you run the `workbench.action.output.toggleOutput` command and:
 - if the bottom panel is hidden, it will show it and show & focus the Output view
 - if the bottom panel is visible, but the Output view is not visible and focused, it will show & focus it
 - if the bottom panel is visible and the Output view is visible & focused, then it will close/hide the bottom panel

I am looking to replicate that behavior with my own views.

Thanks!

**Comment by eamodio** [comment]
@sbatten any idea?

**Comment by alefragnani** [comment]
I wonder if VS Code shouldn't automatically create such command for each _contributed_ view (on panels), just like it already does with the  _Focus on View_ commands (`YourExtension: Focus on YourView View`).

A quick look at VS Code repo, but didn't find any feature request reported.

**Reply by eamodio** [reply]
Yeah that would be nice.

**Reply by eamodio** [reply]
It would be very nice to have an `Extension: Toggle YourView View` command, regardless of placement (not just for the panel). So it could close the sidebar/panel/aux or wherever the view was.

I would also love a way/command to "close" the current view container for a view.

**Comment by isidorn** [comment, marked as answer]
I think this is currently not supported, but is a fair feature request.

**Reply by eamodio** [reply]
@isidorn can this be moved over to the vscode repo then?

**Reply by isidorn** [reply]
Hmm GitHub does not support transfering discussions into issues in another repo unfortunately. Can you please create a new feature request in VS Code? Thank you!

And I will go ahead and close this one.
